kubernetes/client/rest.py

```diff
@@ -155,6 +155,10 @@ def request(self, method, url, query_params=None, headers=None,
                 if query_params:
                     url += '?' + urlencode(query_params)
                 if re.search('json', headers['Content-Type'], re.IGNORECASE):
+                    if headers['Content-Type'] == 'application/json-patch+json':
+                        if not isinstance(body, list):
+                            headers['Content-Type'] = \
+                                'application/strategic-merge-patch+json'
```

Comment:
why is this fix needed?

Reply:
The patch is from https://github.com/kubernetes-client/python/blob/master/scripts/rest_client_patch.diff, which is a old hack for supporting strategic merge patch in the generated client. 

This commit is needed, because the new code generator changed the generated client, and invalidates the old patch. I re-created the patch so the hack works again.

Filed https://github.com/kubernetes-client/python/issues/951 to track adding a test, which ensures strategic merge patch is always supported.